### PR TITLE
feat(todo-12) : Display all tasks in a list

### PR DESCRIPTION
**Pull request by syed-alif-sazim**
**Description of change**
-All tasks created by users will be fetched from localstorage and rendered in the frontend
-completed tasks will appear striked

**Associated trello links**
https://trello.com/c/7pIXLow6

**Screenshot of change**
image

### CODE REVIEW

**Comment by mahir-mohsin-sazim** [comment]
For future PRs, Please follow appropriate commit messages. Follow Github Conventional Commit. And for PR descriptions follow [this](https://docs.google.com/document/d/1mNcOLYRSEUsc9dvmw4V_0xgCAhvjd5PXup6KqdFsT-E/edit?tab=t.0#heading=h.trqjxpk8qcdo)

File: modules/todo/containers/TodoPage/TodoPage.tsx [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+      }
+    }, []);
 
-  return (
+    return (
     <div className="flex items-start justify-center min-h-screen bg-gray-100">
-      <div className="w-full max-w-md bg-white p-6 rounded-lg shadow-md mt-8">
-        <h1 className="text-3xl font-bold text-center text-gray-800 mb-4">To-Do List</h1>
-      </div>
+        <div className="w-full max-w-md bg-white p-6 rounded-lg shadow-md mt-8">
```
Comment (mahir-mohsin-sazim) [inline modules/todo/containers/TodoPage/TodoPage.tsx:25]: NIT: look into tailwind css directives.

File: modules/todo/utils/localStorage.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const loadTasksFromLocalStorage = () => {
+    const storedTasks = localStorage.getItem('tasks');
+    return storedTasks ? JSON.parse(storedTasks) : null;
+  };
+  
+export const saveTasksToLocalStorage = (tasks: any[]) => {
```
Comment (mahir-mohsin-sazim) [inline modules/todo/utils/localStorage.ts:6]: Please avoid using any type

**Review by mahir-mohsin-sazim — CHANGES_REQUESTED** [review]

File: modules/todo/containers/TodoPage/TodoPage.tsx (hunk 1 above)
Comment (syed-alif-sazim) [reply modules/todo/containers/TodoPage/TodoPage.tsx:25]: Bhai where is it? I did not find it in the engineering principles doc.

Comment (mahir-mohsin-sazim) [reply modules/todo/containers/TodoPage/TodoPage.tsx:25]: Its not in our doc. Its in [Tailwind documentation](https://tailwindcss.com/docs/functions-and-directives)

**Review by mahir-mohsin-sazim — APPROVED** [review]

**Pull request merged**